### PR TITLE
Add KV cache to the EAGLE-3 draft head

### PR DESCRIPTION
**Pull request by digantdesai**
Adds a flat KV cache and an explicit-mask attention path (forward_cached) to the
draft head so a proposal step reuses cached keys/values instead of recomputing
the prefix's projections and MLP. Attention still scores against the full
max_seq_len buffer under a static causal mask, matching the target's cache and
keeping the path export-friendly. The stateless is_causal forward is unchanged.

forward_cached requires batch size 1 and contiguous-from-0 writes (overwrites
for speculative rollback are allowed, gapped seeds are not); the invariant is
enforced by an eager-only validator that is skipped under export. Tests cover
cached prefill plus single-step decode against the stateless recompute, rollback
reseeding, and rejection of gapped/offset seeds and B>1.

Authored with assistance from Claude Code.

### CODE REVIEW

**Comment by digantdesai** [comment]
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* #20349
* #20348
* #20347
* #20346
* #20345
* #20344
* #20157
* #20156
* #20155
* #20154
* #20153
* __->__ #20152
* #20151
* #20150
* #20149

File: examples/models/eagle3/draft.py [hunk 1]
```diff
@@ -45,13 +45,56 @@ class Eagle3Config:
...
     has_own_embed: bool = False
+    max_seq_len: int = 4096
 
 
 def _rotate_half(x: torch.Tensor) -> torch.Tensor:
     x1, x2 = x.chunk(2, dim=-1)
     return torch.cat((-x2, x1), dim=-1)
 
 
+class Eagle3KVCache(nn.Module):
```
Comment (metascroy) [inline examples/models/eagle3/draft.py:56]: What distinguishes this from our standard KVCache? Can we just import the standard one?
